keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+
+* The sum of all non-init containers(app and sidecar containers) request/limit for a resource.
+* The highest value resource request or limit among all the init containers.
+
+```
+Effective Request/Limit = Max(Max(Init container limit), Sum(Sidecar containers) + Sum (App Containers)) + Pod Overhead
+```
+
+With the support of pod-level requests and limits, following are a few options that can be used to determine pod’s effective resource request/limit:
+
+* **[Preferred] Option 1: Static values of effective pod request/limit as supplied in the PodSpec.**
+
+  In this the pod level cgroup will be set using the requests and limits from the resources specified at pod-level.
+
+  Example pod spec:
+
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  spec:
+    resources:
+      requests:
+        memory: 100Mi
+        cpu: 1
+      limits:
+        memory: 200Mi
+        cpu: 2
+    containers:
+      - name: container-1
+      - name: container-2
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 1
+  ```
+
+  **Resources requests at pod-level**
+
+  * The scheduler will use the requested resources (CPU and memory) settings specified in the PodSpec to find a suitable node to place the pod. 
+  * Kubelet will use the CPU requests from the pod-level resources section, and translate it to cpu.weight in pod-level cgroup.
+
+  **Resources limits at pod-level**
+
+  * All init(Regular and Sidecar) and non-init containers will be capped by the pod level limit which represents the peak usage of all containers.
+  * Kubelet will use CPU and memory limits from PodSpec, and translate them to cpu.max and memory.max cgroup settings at pod-level respectively. 
+
+  **Pros**
+
+  * Simple to understand and implement.
+  * Guarantees enough resources for all containers.
+
+  **Cons**
+
+  * Can lead to over-provisioning (resource wastage), especially if the init container peak usage is significantly higher than the steady-state usage 
+  of the other containers (sidecar and app containers). However this can be mitigated by setting container level resource spec for this case.
+  
+  Example 1: If your application containers require high resources simultaneously, set pod-level requests/limits such that it is an aggregate of all container-level requests/limits.
+
+  Example 2: If your application has init containers (and not sidecar containers) with high resource requirements and usage, while your application containers require fairly less resources, 
+  set pod-level requests/limits as maximum of init and non-init container resources requirements. 
+
+  Example 3: For a Pod with init containers, sidecar containers and app containers, set the pod level requests/limits such that:
+  
+  ```
+  Pod level Request/Limit = Max(Max(Init container limit), Sum(Sidecar containers) + Sum (App Containers))
+  ```
+
+* **Option 2: Phased requests/limits - Pod-level cgroups are set after regular init containers are completed.**
```

Comment:
if we do not do it, perhaps we can delete this section to avoid confusion. Or we are still undecided?